#! /bin/sh /usr/share/dpatch/dpatch-run
## 41_maketexmf.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix for mktexmf script from perforce depot

@DPATCH@
 build/source/texk/kpathsea/mktexmf |   19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

Index: texlive-bin-2006.svn3816/build/source/texk/kpathsea/mktexmf
===================================================================
--- texlive-bin-2006.svn3816.orig/build/source/texk/kpathsea/mktexmf	2006-01-17 22:41:51.000000000 +0100
+++ texlive-bin-2006.svn3816/build/source/texk/kpathsea/mktexmf	2007-01-26 04:01:50.000000000 +0100
@@ -36,17 +36,20 @@
   rootfile=$sauterroot
   rootname=b-$rootname
 else
-  case $rootname in
+  case $name in
     # csso12[0-5][0-9] from CJK package; unsupported by mktexmf
     csso12[0-5][0-9])
-       rootfile=;;
-    cs*|lcsss*|icscsc*|icstt*|ilcsss*)
-       rootfile=`kpsewhich cscode.mf`;;
-    wn[bcdfirstuv]*|rx[bcdfiorstuvx][bcfhilmostx]|l[abcdhl][bcdfiorstuvx]*)
-       lhprefix=`echo "$name" | sed 's/^\(..\).*/\1/'`
-       rootfile=`kpsewhich ${lhprefix}codes.mf 2>/dev/null`;;
+      rootfile=;;
     *)
-       rootfile=`kpsewhich $rootname.mf 2>/dev/null`;;
+      case $rootname in
+        cs*|lcsss*|icscsc*|icstt*|ilcsss*)
+          rootfile=`kpsewhich cscode.mf`;;
+        wn[bcdfirstuv]*|rx[bcdfiorstuvx][bcfhilmostx]|l[abcdhl][bcdfiorstuvx]*)
+          lhprefix=`echo "$name" | sed 's/^\(..\).*/\1/'`
+          rootfile=`kpsewhich ${lhprefix}codes.mf 2>/dev/null`;;
+        *)
+          rootfile=`kpsewhich $rootname.mf 2>/dev/null`;;
+      esac;;
   esac
 fi
 { test -z "$rootfile" || test ! -f "$rootfile"; } && exit 1
